### PR TITLE
Add interface for Serializer classes

### PR DESCRIPTION
**Pull request by nnatter**
This change makes it easier to overwrite a Serializer class in a project. For example, overwriting a Serializer class is necessary if a project extends the `Media` entity.

### CODE REVIEW

File: Content/Serializer/MediaSerializerInterface.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+use JMS\Serializer\SerializationContext;
+use Sulu\Bundle\MediaBundle\Api\Media;
+
+interface MediaSerializerInterface
+{
+    /**
+     * @return mixed[]
+     */
+    public function serialize(Media $media, ?SerializationContext $context = null): array;
```
Comment (alexander-schranz) [inline Content/Serializer/MediaSerializerInterface.php:24]: Should be `MediaInterface`

File: Content/Serializer/ContactSerializerInterface.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+use JMS\Serializer\SerializationContext;
+use Sulu\Bundle\ContactBundle\Api\Contact;
+
+interface ContactSerializerInterface
+{
+    /**
+     * @return mixed[]
+     */
+    public function serialize(Contact $contact, string $locale, ?SerializationContext $context = null): array;
```
Comment (alexander-schranz) [inline Content/Serializer/ContactSerializerInterface.php:24]: should be `ContactInterface`

File: Content/Serializer/CategorySerializerInterface.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+use JMS\Serializer\SerializationContext;
+use Sulu\Bundle\CategoryBundle\Api\Category;
+
+interface CategorySerializerInterface
+{
+    /**
+     * @return mixed[]
+     */
+    public function serialize(Category $category, ?SerializationContext $context = null): array;
```
Comment (alexander-schranz) [inline Content/Serializer/CategorySerializerInterface.php:24]: Should be `CategoryInterface`

File: Content/Serializer/AccountSerializerInterface.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+use JMS\Serializer\SerializationContext;
+use Sulu\Bundle\ContactBundle\Api\Account;
+
+interface AccountSerializerInterface
+{
+    /**
+     * @return mixed[]
+     */
+    public function serialize(Account $account, string $locale, ?SerializationContext $context = null): array;
```
Comment (alexander-schranz) [inline Content/Serializer/AccountSerializerInterface.php:24]: Should be `AccountInterface`

**Comment by alexander-schranz** [comment]
I basically don't see that this should be the recommended way to override. How the media should be serialized. 

Best would be if we could configure `<params>` on the content types for serialization groups.

File: Content/Serializer/MediaSerializerInterface.php (hunk 1 above)
Comment (nnatter) [reply Content/Serializer/MediaSerializerInterface.php:24]: no, the serializer expect an `ApiEntitiy` instance at the moment: https://github.com/sulu/SuluHeadlessBundle/issues/21

**Comment by nnatter** [comment]
This classes are not used for content types only. For example, the `MediaSerializer` class is also used  for serializing the `Media` of a navigation item or the avatar of a `Contact`. 🙂

**Comment by alexander-schranz** [comment]
@nnatter yeah but that would be configured on the `contact_selection` how the contact should serialized its avatar. And for all hardcoded calls of the serializer (The only one I could think about is the `author`) this groups should be configureable over the headless bundle.

**Pull request merged**